malaria-ios/ViewControllers/UserProfileViewController.swift

```diff
@@ -0,0 +1,199 @@
+    view.backgroundColor = UIColor(patternImage: UIImage(named: "background")!)
+    
+    // Get the last reminder value from the User Defaults
+    let value = UserSettingsManager.UserSetting.PillReminderValue.getString()
+    
+    remindMeWeeksButton.setTitle(value, forState: .Normal)
+
+    self.reminderValue = value.toReminderValue()
+  }
+  
+  override func viewWillAppear(animated: Bool) {
+    super.viewWillAppear(animated)
+    
+    context = CoreDataHelper.sharedInstance.createBackgroundContext()!
+    medicineManager = MedicineManager(context: context!)
+    psnm = PillStatusNotificationsManager(context: context!)
+    
+    getRegisteredMedicine()
+    calculateMedicineLeft()
+    recalculateTableHeight()
+  }
+  
+  func dismissKeyboard() {
+    view.endEditing(true)
+  }
+  
+  func getRegisteredMedicine() {
+    medicine = []
+    medicine.append(medicineManager!.getCurrentMedicine()!)
+    
+    // To get all medicine use only:
+    // medicine = medicineManager!.getRegisteredMedicines()
+  }
+  
+  func calculateMedicineLeft() {
+    guard let currentMedicine = medicineManager!.getCurrentMedicine() else {
+      remainingLabel.text = ""
+      return
+    }
+    
+    tableView.reloadData()
+
+    // remaining value is floored (opposed to ceiled)
+    let remaining = Int(currentMedicine.remainingMedicine) / currentMedicine.interval
+    
+    let timeMeasuringUnit = (currentMedicine.interval == 7) ? "weeks" : "days"
+    
+    remainingLabel.text = "You have \(remaining) \(timeMeasuringUnit) of \(currentMedicine.name) left."
+    
+    // Save pill status to show it in the widget
+    WidgetSettingsManager.WidgetSetting.RemainingPills.setObject(remaining)
+    WidgetSettingsManager.WidgetSetting.RemainingPillsMeasuringUnit.setObject(timeMeasuringUnit)
+    
+    let shouldPresentNotification: Bool = psnm!.shouldPresentNotification(remaining, reminderValue: reminderValue.rawValue)
+    
+    shouldPresentNotification ? psnm!.scheduleNotification(NSDate()) : psnm!.unsheduleNotification()
+    
+    let defaultBrownTint = self.remindMeWeeksButton.titleLabel?.textColor
+    self.remainingLabel.textColor = shouldPresentNotification ? UIColor.redColor() : defaultBrownTint
+  }
+  
+  func recalculateTableHeight() {
+    tableViewHeightConstraint.constant = CGFloat(medicine.count) * Constants.UserProfile.CellHeightAndOffset
+  }
+}
+
+// MARK: IBActions and helpers
+
+extension UserProfileViewController {
+  
+  @IBAction func settingsBtnHandler(sender: AnyObject) {
+    dispatch_async(dispatch_get_main_queue()) {
+      let view = UIStoryboard.instantiate(viewControllerClass: SetupScreenViewController.self)
+      view.delegate = self
+      self.presentViewController(view, animated: true, completion: nil)
+    }
+  }
+  
+  @IBAction func remindMeWeeksPressed(sender: UIButton) {
+    let alertController = UIAlertController(title: "Set reminder for:", message: nil, preferredStyle: UIAlertControllerStyle.Alert)
```

Comment:
Move "magic" strings and numbers (this case just string) to a dedicated constant similar to: 
https://github.com/systers/malaria-app-ios/blob/develop/malaria-ios/ViewControllers/PlanTrip/PlanTripViewController.swift#L305

In the future we might need to Localize the strings therefore moving the strings from the UIAlertController is a necessary step.